de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_Extend.java

```diff
@@ -71,27 +75,23 @@ protected String doIt() throws Exception
 
 			addLog("@Processed@: @C_Flatrate_Term_ID@ " + termToExtend.getC_Flatrate_Term_ID());
 			InterfaceWrapperHelper.save(termToExtend);
-			
+
 			getResult().setRecordToRefreshAfterExecution(TableRecordReference.of(termToExtend));
 		}
 		else
 		{
-			// extend all terms that are due for extension
-			final String wc = "COALESCE (" + I_C_Flatrate_Term.COLUMNNAME_AD_PInstance_EndOfTerm_ID + ",0)=0 AND "
-					+ I_C_Flatrate_Term.COLUMNNAME_DocStatus + "='" + X_C_Flatrate_Term.DOCSTATUS_Completed + "' AND "
-					+ I_C_Flatrate_Term.COLUMNNAME_NoticeDate + "<? AND "
-					// 04432 don't extend canceled contracts
-					+ "COALESCE (" + I_C_Flatrate_Term.COLUMNNAME_ContractStatus + ",'') != " + DB.TO_STRING(X_C_Flatrate_Term.CONTRACTSTATUS_Quit);
-
-			final Iterator<I_C_Flatrate_Term> termsToExtend = new Query(getCtx(), I_C_Flatrate_Term.Table_Name, wc, get_TrxName())
-					.setParameters(SystemTime.asTimestamp())
-					.setClient_ID() // .setApplyAccessRules requires a more sophisticated role access setup which we urrently don't have to the user "system" () and this is not a "user"
-					// process
+			final Iterator<I_C_Flatrate_Term> termsToExtend = queryBL.createQueryBuilder(I_C_Flatrate_Term.class)
+					.addInArrayFilter(I_C_Flatrate_Term.COLUMNNAME_AD_PInstance_EndOfTerm_ID, 0, null)
+					.addEqualsFilter(I_C_Flatrate_Term.COLUMN_DocStatus, IDocument.STATUS_Completed)
+					.addCompareFilter(I_C_Flatrate_Term.COLUMN_NoticeDate, Operator.LESS, SystemTime.asTimestamp())
+					.addNotInArrayFilter(I_C_Flatrate_Term.COLUMN_ContractStatus, Arrays.asList(X_C_Flatrate_Term.CONTRACTSTATUS_Quit, X_C_Flatrate_Term.CONTRACTSTATUS_Voided))
+					.orderBy().addColumn(I_C_Flatrate_Term.COLUMN_C_Flatrate_Term_ID).endOrderBy()
+					.create()
+					.setClient_ID()
```

Comment:
not now... next time pls use org.adempiere.ad.dao.IQueryBuilder.addOnlyContextClient()